cf-agent/package_module.c

```diff
@@ -0,0 +1,1611 @@
+            }
+            else if (StringSafeEqual(type, "repo"))
+            {
+                package_data->type = PACKAGE_TYPE_REPO;
+            }
+            else
+            {
+                Log(LOG_LEVEL_VERBOSE, "unsupported package type: %s", type);
+                free(package_data);
+                return NULL;
+            }
+        }
+        else if (StringStartsWith(line, "Name="))
+        {
+            if (package_data->name)
+            {
+                /* Some error occurred as we already have name for 
+                 * given package. */
+                Log(LOG_LEVEL_ERR, "Duplicated package name "
+                    "received: [%s] %s", line, package_data->name);
+                free(package_data);
+                return NULL;
+            }
+            package_data->name = 
+                SafeStringDuplicate(line + strlen("Name="));
+        }
+        else if (StringStartsWith(line, "Version="))
+        {
+            if (package_data->version)
+            {
+                /* Some error occurred as we already have version for 
+                 * given package. */
+                Log(LOG_LEVEL_ERR, "Duplicated package version "
+                    "received: [%s] %s", line, package_data->version);
+                free(package_data);
+                return NULL;
+            }
+            package_data->version = 
+                SafeStringDuplicate(line + strlen("Version="));
+        }
+        else if (StringStartsWith(line, "Architecture="))
+        {
+            if (package_data->arch)
+            {
+                /* Some error occurred as we already have arch for 
+                 * given package. */
+               Log(LOG_LEVEL_ERR, "Duplicated package architecture "
+                    "received: [%s] %s", line, package_data->arch);
+                free(package_data);
+                return NULL;
+            }
+            package_data->arch = 
+                SafeStringDuplicate(line + strlen("Architecture="));
+        }
+        /* For handling errors */
+        else if (StringStartsWith(line, "Error="))
+        {
+            Log(LOG_LEVEL_VERBOSE, "have error: %s",
+                line + strlen("Error="));
+        }
+        else if (StringStartsWith(line, "ErrorMessage="))
+        {
+            Log(LOG_LEVEL_VERBOSE, "have error message: %s",
+                line + strlen("ErrorMessage="));
+        }
+        else
+        {
+            Log(LOG_LEVEL_VERBOSE, "unsupported option: %s", line);
+        }
+    }
+    
+    return package_data;
+}
+
+static int NegotiateSupportedAPIVersion(PackageModuleWrapper *wrapper)
+{
+    int api_version = -1;
+
+    Rlist *response = NULL;
+    if (ReadWriteDataToPackageScript("supports-api-version", "",
+                                     &response, wrapper) != 0)
+    {
+        Log(LOG_LEVEL_INFO, "Some error occurred while communicating with "
+            "wrapper while getting supported API version.");
```

Comment:
Fixed! General error is printed with this log message 'Log(LOG_LEVEL_ERR,
+ "Can not set up wrapper for module: %s", pm_name);'. This is just more detailed and can be seen once we are interested in more details (info level in this case).